pkg/controller/common/annotation/controller_version.go

```diff
@@ -73,6 +73,11 @@ func ReconcileCompatibility(ctx context.Context, client k8s.Client, obj ctrlclie
 
 	annotatedVersion := obj.GetAnnotations()[ControllerVersionAnnotation]
 
+	_, err := version.Parse(controllerVersion)
+	if err != nil {
+		return false, errors.Wrap(err, "Error parsing controller version")
+	}
+
```

Comment:
Like @barkbay proposed in the inital issue, I think we should only attempt to parse the controller version before we set it in the resource annotation (before L45 in UpdateControllerVersion):

https://github.com/elastic/cloud-on-k8s/blob/1f8464a90f495368a00c8264cbaf5148ab645964/pkg/controller/common/annotation/controller_version.go#L45

It will be good to rename the variable `version` in `controllerVersion` to avoid a conflict with the `version` package used to call the function `Parse`.